backend/main.go

```diff
@@ -441,6 +441,7 @@ func main() {
 			zapier.CreateZapierRoutes(r, db, &zapierStore, &rh)
 		})
 		r.HandleFunc("/slack-events", privateResolver.SlackEventsWebhook(ctx, slackSigningSecret))
+		r.Post(fmt.Sprintf("%s/%s", privateEndpoint, "login"), privateResolver.Login)
```

Comment:
Was going to put this on line 445 just before setting up the PrivateMiddleware.
Getting an error from chi that we can't set middleware before a route is setup.

This is nor readable though and it doesn't feel right to some extent.
The endpoint still works if we put it on line 445 but if someone
accidentally hits the login endpoint with a token or similar "things"
that are validated by the PrivateMiddleware, they will not be able to login.
So trade-offs perhaps?

Reference: https://github.com/go-chi/chi/issues/125

Reply:
As I explained in my original comment, the chi library prevents us from doing that because of its limitations on what can be done inside an ``r.Route function``. According to the [referenced](https://github.com/go-chi/chi/issues/125) github issue, we can't do the following:

```go
r.Route(privateEndpoint, func(r chi.Router) {
    r.Post("/login", privateResolver.Login)
    r.Use(private.PrivateMiddleware)
    r.Use(highlightChi.Middleware) 
}
```

Because we set up a route before setting up middleware via the ```r.Use```.
I tried it and there was an compiling the source.

I'm open to ideas here because our solution is now dependent on our preferred
"style". I know it doesn't look pretty but it kinda works also.

Reply:
ahh got it! no worries then, this is fine as is. thank you for the detailed explanation and the reference to the original discussion on go-chi